### ISSUE TITLE
Design roadmap for the annotation tool

### ISSUE THREAD

**Issue by constantinpape** [hci-unihd/covid-if-annotations#1]
The goal of our proofreading efforts is to generate ground-truth for cell segmentation and infected / control cell classification.
@akreshuk, @wolny and me discussed how to design the tool for this today:
1. In the first step, users correct the existing segmentation using the painting functionality of the labels layer.
2. In the second step, users label the cells into "infected", "control" or "uncertain".

For the first step, having functionality to hide the cells that have already been inspected / proofread would be very helpful. I will open a separate issue to discuss this in more detail.

For the second step users would start from scratch (or from previous partial annotations, see below) and would need to label all cells as belonging to one of the three categories. In my prototype, I have done this by adding an additional labels layer that was initially empty and then propagating the annotations from this layer to the cell labels, and displaying the results in a separate layer that outlines the edges. See this image:

![Screenshot from 2020-05-21 14-39-23](https://user-images.githubusercontent.com/4263537/82711241-17142100-9c85-11ea-8af5-2f8117e923a2.png)

Once both steps are completed users upload the exported results via the EMBL annotation app (which is also used to obtain the initial files).

Also, we would like to allow users to submit "partially" annotated images. An image could be marked as partial if significant segmentation errors were noticed after step 1 was already finished (we don't want to allow going back and forth between step 1 and 2 in a single session to keep things simple) or if not all cells were classified in step 2.
Partial images can be uploaded in the same way, but are marked differently and would go back into the pool of available images for annotation; but initialized with the partially corrected segmentation / partially classified cells.

**Comment by constantinpape** [comment]
@sofroniewn [suggested to use a points layer for the infected / control classification instead of a labels layer](https://github.com/hci-unihd/covid-if-annotations/issues/2#issuecomment-632975634).

I think this makes a lot of sense, and it might make switching between the segmentation correction and semantic classification so easy that we don't need to strictly separate step 1 and 2.

**Comment by constantinpape** [comment]
Ok, this is very quickly taking shape. I am using the `points_layer` for annotations now, I think this is very convenient in terms of UI and with it we don't need a strict separation between segmentation correction / semantic annotations.
I think what's left to do for the annotation tool part is:
- [ ] Implement hiding annotated objects using a fixed color map (if this is possible, see #5).
- [x] Fix annotations via mouse clicking, see #2. 
- [x] Use `magicgui` to also register the keys for updating layers `u` and toggling hiding `h` / toggling the annotation mode `t` with the gui.
- [x] Implement export of the annotation results.

and some minor clean up / implementation details.

**Comment by constantinpape** [comment]
All the initial functionality we have is implemented now, except the full label hiding functionality, but this is pending upstream changes, see #5.
We are waiting for some test user feedback and will announce it "officially" to users on Tuesday.
